### ISSUE TITLE
Refactor emitInsBinary to be mor readable and remove check duplication

### ISSUE THREAD

**Issue by tannergooding** [dotnet/coreclr#15829]
[emitInsBinary](https://github.com/dotnet/coreclr/blob/master/src/jit/emitxarch.cpp#L2902) currently duplicates a lot of checks, is hard to follow, and could (overall) use a good refactoring.

An example of this duplication is how it does a `isContained` check 5 times for `src` and 2 times for `dst` in the first 90 lines of the method. It also does a `isUsedFromSpillTemp` check 3 times for `src` and 2 times for `dst`.

It also splits the analysis for a particular node type out into multiple places in the method, rather than keeping it contained and ends up doing checks even when a previous check may have invalidated the latter from being possible.

**Comment by RussKeldorph** [comment]
@dotnet/jit-contrib

**Comment by tannergooding** [comment]
A PR that attempts to resolve the issue is here: https://github.com/dotnet/coreclr/pull/15836
